tests/unit/test_unit_smhi.py

```diff
@@ -91,50 +91,37 @@ def test_unit_get_data_by_city(
         assert data == "test"
 
     @pytest.mark.parametrize(
-        "stations,latitude, longitude,dist",
+        "latitude, longitude,dist",
         [
             (
-                MagicMock(
-                    **{
-                        "id": [1, 2],
-                        "name": ["Akalla", "Högdalen"],
-                        "latitude": [59.5, 59.3],
-                        "longitude": [17.8, 17.8],
-                    }
-                ),
                 59,
                 17,
                 None,
             ),
             (
-                MagicMock(
-                    **{
-                        "id": [1, 2],
-                        "name": ["Akalla", "Högdalen"],
-                        "latitude": [59.5, 59.3],
-                        "longitude": [17.8, 17.8],
-                    }
-                ),
                 59.4,
                 17,
                 30,
             ),
         ],
     )
     @patch("geopy.distance")
-    def test_find_stations_from_gps(
-        self, mock_distance, stations, latitude, longitude, dist
-    ):
+    def test_find_stations_from_gps(self, mock_distance, latitude, longitude, dist):
         """Unit test for SMHI find_stations_from_gps method.
 
         Args:
         """
-
+        stations = MagicMock()
```

Comment:
This mock contains a `station` field that isn't an iterable, so in `_find_stations_from_gps` the list comprehensions will return an empty list and you will have 

`nearby_town = []`, which is what the test complains about

```
>       assert nearby_town[0][0] in stations.station.id
E       IndexError: list index out of range
```

I'm not sure if that will ever be possible in the real code, but it could be good to catch that case.